server/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -292,9 +295,20 @@ public Task unregister(Task task) {
             }
         } finally {
             tracer.stopTrace("task-" + task.getId());
+            for (RemovedTaskListener listener : removedTaskListeners) {
+                listener.onRemoved(task);
+            }
         }
     }
 
+    public void addRemovedTaskListener(RemovedTaskListener removedTaskListener) {
```

Comment:
nit: can we rename this listener to `UnregisterTaskListener`? the removeRemoved... it's a bit confusing.

Reply:
Absolutely! This also was irking me for a while...